### PR TITLE
translation: Add the contributing guidelines for Chinese-to-English

### PR DESCRIPTION
**Pull request by krahets**
- Background requirements for contributors.
- Translation standards.
- Translation priorities of contributors from different linguistic backgrounds.
- Detailed guidance on the translation process.

### CODE REVIEW

**Review by pengchzn — APPROVED** [review]
The document is excellently written, with a very clear and understandable structure. The pseudo-code offers clear procedural guidance for both translators and reviewers, it will be helpful. I look forward to collaborating on the translation!

File: en/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Contributing guidelines for Chinese-to-English
+
+We are working on translating "Hello Algo" from Chinese to English with the following approach:
+
+1. **AI translation**: Carry out an initial pass of translations using the machine learning translator.
+2. **Human optimization**: Manually refine the machine-generated outputs to ensure authenticity and accuracy.
+3. **Pull request review**: The optimized translation will be doubly checked by the reviewers through GitHub pull request workflow.
```
Comment (RafaelCaso) [inline en/CONTRIBUTING.md:7]: I think "double checked" is better ("The optimized translation will be double checked by the reviewers through GitHub pull request workflow

File: en/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+We're seeking contributors who meet the following criteria.
+
+- **Technical background**: Strong foundation in computer science, particularly in data structures and algorithms.
+- **Language skills**: Native proficiency in Chinese with professional-level English, or native English.
+- **Available time**: Dedicated to contributing to open-source projects with a willingness to engage in long-term translation efforts.
+
+That is, our contributors are computer scientists, engineers, and students from different linguistic backgrounds, and their objectives have different focal points:
+
+- **Native Chinese with professional working English**: Ensuring translation accuracy and consistency between CN and EN versions.
+- **Native English**: Enhance the authenticity and fluency of the English content to read natural and engaging.
```
Comment (RafaelCaso) [inline en/CONTRIBUTING.md:21]: 'natural' needs to be conjugated. I suggest "Enhance the authenticity and fluency of the English content to read naturally and to be engaging"

File: en/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+> The "Accuracy" and "Authenticity" are primarily handled by native Chinese speakers and native English speakers, respectively.
+
+**Accuracy**:
+
+- Maintain consistency in terminology across translations by referring to the [Terminology](https://www.hello-algo.com/chapter_appendix/terminology/) section.
+- Prioritize technical accuracy and maintain the tone and style of the Chinese version. If you think changing the original meaning is necessary, please first submit a PR to modify the Chinese version.
+
+**Authenticity**:
+
+- Translations should read natural and fluent, adhering to English expression conventions.
```
Comment (RafaelCaso) [inline en/CONTRIBUTING.md:46]: 'natural' and 'fluent' should be adverbs (they describe the verb 'read'). I suggest "Translations should read naturally and fluently, adhering to English expression conventions."

**Review by RafaelCaso — CHANGES_REQUESTED** [review]
There are two changes that I feel strongly about on Lines 21 and 46. There is a stylistic suggestion for line 7. (I've certainly heard 'doubly checked' before but it feels weird to me. I'd be curious what other reviewers think). I absolutely love the pseudocode by the way.

File: en/CONTRIBUTING.md (hunk 1 above)
Comment (krahets) [reply en/CONTRIBUTING.md:7]: Fixed. Thanks!

**Comment by krahets** [comment]
Thanks @RafaelCaso ! I think pseudocode can cleanly show the process details and actions. It is like the text version of a flowchart.

**Review by RafaelCaso — APPROVED** [review]

**Review by RafaelCaso — APPROVED** [review]

**Comment by thomasq0** [comment]
Well written guideline.

I understand both accuracy, honest translation from Chinese version, and authenticity are important for this translation work. But I find English a better language than Chinese when it comes to delivering the exact meaning. Sometimes, a same Chinese word may imply very different things/concepts in different context. According to the pseudocode, it appears to me that authenticity triumphs if the expression looks "ok" but not clear enough for native readers. Is it right?

**Comment by krahets** [comment]
> But I find English a better language than Chinese when it comes to delivering the exact meaning. Sometimes, a same Chinese word may imply very different things/concepts in different context.

Agree on that.

> it appears to me that authenticity triumphs if the expression looks "ok" but not clear enough for native readers. Is it right?

Yep. Because it is translated from Chinese, I believe there will be some unnatural expressions for native readers. Without compromising accuracy, we can choose to optimize authenticity.

File: en/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+# Contributing guidelines for Chinese-to-English
+
+We are working on translating "Hello Algo" from Chinese to English with the following approach:
+
+1. **AI translation**: Carry out an initial pass of translations using the machine learning translator.
```
Comment (yuelinxin) [inline en/CONTRIBUTING.md:5]: "the machine learning translator" -> "large language models".

File: en/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+We're seeking contributors who meet the following criteria.
+
+- **Technical background**: Strong foundation in computer science, particularly in data structures and algorithms.
+- **Language skills**: Native proficiency in Chinese with professional-level English, or native English.
+- **Available time**: Dedicated to contributing to open-source projects with a willingness to engage in long-term translation efforts.
+
+That is, our contributors are computer scientists, engineers, and students from different linguistic backgrounds, and their objectives have different focal points:
+
+- **Native Chinese with professional working English**: Ensuring translation accuracy and consistency between CN and EN versions.
+- **Native English**: Enhance the authenticity and fluency of the English content to read naturally and to be engaging.
```
Comment (yuelinxin) [inline en/CONTRIBUTING.md:21]: "read naturally" -> "flow naturally".

File: en/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+> The "Accuracy" and "Authenticity" are primarily handled by native Chinese speakers and native English speakers, respectively.
+
+**Accuracy**:
+
+- Maintain consistency in terminology across translations by referring to the [Terminology](https://www.hello-algo.com/chapter_appendix/terminology/) section.
+- Prioritize technical accuracy and maintain the tone and style of the Chinese version. If you think changing the original meaning is necessary, please first submit a PR to modify the Chinese version.
+
+**Authenticity**:
+
+- Translations should read naturally and fluently, adhering to English expression conventions.
```
Comment (yuelinxin) [inline en/CONTRIBUTING.md:46]: "read naturally" -> "flow naturally".

File: en/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+- Each PR should cover at least one complete document to ensure manageable review sizes, except for bug fixes.
+
+**Review**:
+
+- During the review, prioritize evaluating the changes, consulting the surrounding context as needed.
+- Learning from each other's perspectives can lead to better translations and more cohesive results.
+
+## Translation pseudo-code
+
+The following pseudo-code demonstrates a typical translation approach to translators.
```
Comment (yuelinxin) [inline en/CONTRIBUTING.md:62]: The following pseudo-code models the steps in a typical translation process.

File: en/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        if is_authentic_English(sentence):
+            continue
+        # Optimize the authenticity
+        result = refine_authenticity(sentence)
+        # Add comments in the PR if it may break consistency
+        if break_consistency(result):
+            add_comment(description)
+
+    pull_request = submit_pull_request(markdown_texts)
+    # The PR will be merged after approved by >= 2 reviewers
```
Comment (yuelinxin) [inline en/CONTRIBUTING.md:88]:
```
if count_approved_reviews(pull_request) >= 2:
    merge(pull_request)
else:
    modify(pull_request)
```

**Review by yuelinxin — COMMENTED** [review]
Just a few wording suggestions, great work!

**Comment by QiLOL** [comment]
Looks great! Thank you!

File: en/CONTRIBUTING.md (hunk 8 above)
Comment (krahets) [reply en/CONTRIBUTING.md:88]: Good modification! I prefer:

```py
while count_approvals(pull_request) < 2:
          continue
merge(pull_request)
```

**Comment by krahets** [comment]
Thanks for the great review!

**Pull request merged**